### ISSUE TITLE
DefaultTagUserApi.getTagDefinitionForName throws NPE if tagDefinitionName doesn't exist

### ISSUE THREAD

**Issue by SquareMike** [killbill/killbill#822]
While experimenting with the Tag Api we saw that an NPE will be thrown if you call DefaultTagUserApi.getTagDefinitionForName with a tagDefinitionName that has not been defined yet.  If the system threw something like "TagDefinitionDoesNotExistException" then the client experience would be better.

Stack:
```
java.lang.NullPointerException
    at org.killbill.billing.util.tag.DefaultTagDefinition.<init>(DefaultTagDefinition.java:39)
    at org.killbill.billing.util.tag.api.DefaultTagUserApi.getTagDefinitionForName(DefaultTagUserApi.java:188)
    at org.killbill.billing.util.glue.KillbillApiAopModule$ProfilingMethodInterceptor$1.execute(KillbillApiAopModule.java:52)
    at org.killbill.commons.profiling.Profiling.executeWithProfiling(Profiling.java:33)
    at org.killbill.billing.util.glue.KillbillApiAopModule$ProfilingMethodInterceptor.invoke(KillbillApiAopModule.java:49)
```

Kill Bill Version: 0.18.14

Thank you!

**Comment by sbrossie** [comment]
@SquareMike PR gladly accepted ;-)

**Comment by SquareMike** [comment]
Great, thanks for reviewing @sbrossie .

**Comment by sbrossie** [comment]
Fixed in c07541e43439be2517d532e9653e0073ced8417b
